src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -76,6 +54,65 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+
+        internal async Task<HttpResponseMessage> CreateCheckStatusResponse(
+            HttpRequestMessage request,
+            string instanceId,
+            OrchestrationClientAttribute attribute,
+            int totalTimeout,
+            int retryTimeout)
+        {
+            GetClientResponseLinks(request, instanceId, attribute, out var statusQueryGetUri, out var sendEventPostUri,
+                out var terminatePostUri);
+
+            if (totalTimeout < retryTimeout)
+            {
+                throw new InvalidOperationException($"Total timeout {totalTimeout} should be bigger than {retryTimeout}");
+            }
+
+            var iterationCount = totalTimeout / retryTimeout;
+            JToken durableFunctionOutput = null;
+
+            var client = GetClient(request);
+            var status = await client.GetStatusAsync(instanceId);
+            for (var i = 0; i < iterationCount; i++)
+            {
+                Thread.Sleep(retryTimeout);
+                if (status != null && status.RuntimeStatus == OrchestrationRuntimeStatus.Completed)
+                {
+                    durableFunctionOutput = status.Output;
+                    break;
+                }
+                status = await client.GetStatusAsync(instanceId);
+            }
+
+            if (durableFunctionOutput == null)
+            {
+                var response = request.CreateResponse(
+                    HttpStatusCode.Accepted,
+                    new
+                    {
+                        id = instanceId,
+                        statusQueryGetUri,
+                        sendEventPostUri,
+                        terminatePostUri
+                    });
+
+                // Implement the async HTTP 202 pattern.
+                response.Headers.Location = new Uri(statusQueryGetUri);
+                response.Headers.RetryAfter = new RetryConditionHeaderValue(TimeSpan.FromSeconds(10));
+                return response;
+            }
+            else
+            {
+                var response = request.CreateResponse(
+                    HttpStatusCode.OK, durableFunctionOutput);
```

Comment:
nit: move these parameters up to be on the same line as the method call.
`var response = request.CreateResponse(HttpStatusCode.OK, durableFunctionOutput);`